echopype/qc/api.py

```diff
@@ -9,32 +9,34 @@
 logger = _init_logger(__name__)
 
 
-def _clean_ping_time(ping_time_old, local_win_len=100):
-    ping_time_old_diff = np.diff(ping_time_old)
-    neg_idx = np.argwhere(
-        ping_time_old_diff < np.timedelta64(0, "ns")
-    )  # indices with negative diff
-    if neg_idx.size != 0:
-        ni = neg_idx[0][0]
-        local_win = np.arange(-local_win_len, local_win_len)
-        local_pt_diff = ping_time_old_diff[ni + local_win]
-        local_pt_median = np.median(
-            np.delete(local_pt_diff, local_win_len)
-        )  # median after removing negative element
-        ping_time_new = np.hstack(
-            (
-                ping_time_old[: ni + 1],
-                ping_time_old[ni]
-                + np.cumsum(np.hstack((local_pt_median, ping_time_old_diff[(ni + 1) :]))),
-            )
-        )
-        return _clean_ping_time(ping_time_new, local_win_len=local_win_len)
-    else:
-        return ping_time_old  # no negative diff
+def _clean_reversed(time_old: np.ndarray, win_len: int):
+    time_old_diff = np.diff(time_old)
+
+    # get indices of arr_diff with negative values
+    neg_idx = np.argwhere(time_old_diff < np.timedelta64(0, "ns")).flatten()
+
+    # substitute out the reversed timestamp using the previous one
+    new_diff = []
+    for ni in neg_idx:
+        local_win_idx = ni + np.arange(-win_len, 0)
+        if local_win_idx[0] < 0:
+            first_valid_idx = np.argwhere(local_win_idx == 0).flatten()[0]
+            local_win_idx = local_win_idx[first_valid_idx:]
+        new_diff.append(np.median(time_old_diff[local_win_idx]))
+    time_old_diff[neg_idx] = new_diff
+
+    # perform cumulative sum of differences after 1st neg index
+    c_diff = np.cumsum(time_old_diff[neg_idx[0] :], axis=0)
+
+    # create new array that preserves differences, but enforces increasing vals
+    new_time = time_old.copy()
+    new_time[neg_idx[0] + 1 :] = new_time[neg_idx[0]] + c_diff
+
+    return new_time
 
 
 def coerce_increasing_time(
-    ds: xr.Dataset, time_name: str = "ping_time", local_win_len: int = 100
+    ds: xr.Dataset, time_name: str = "ping_time", win_len: float = 100
```

Comment:
Same type change as in `_clean_reversed()`
```suggestion
    ds: xr.Dataset, time_name: str = "ping_time", win_len: int = 100
```